elide-async/src/main/java/com/yahoo/elide/async/service/AsyncExecutorService.java

```diff
@@ -133,10 +127,12 @@ public void executeQuery(AsyncQuery queryObj, User user, String apiVersion) {
      * @param user User
      * @param apiVersion API Version
      */
-    public void completeQuery(AsyncQuery query, User user, String apiVersion) {
-        if (query.getQueryUpdateWorker() != null) {
+    public void completeQuery(AsyncAPI query, User user, String apiVersion) {
+        AsyncAPIUpdateThread queryUpdateWorker = futureTask.get();
+        if (queryUpdateWorker != null) {
```

Comment:
It won't be safe to use null/not-null here to signify anything.  After the first threadlocal is initialized by a timeout, it will be non-null forever thereafter.